### PR TITLE
Case 20738: add dynamic multiselect for material target property

### PR DESCRIPTION
**Pull request by dback2**
https://highfidelity.fogbugz.com/f/cases/20738/

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18588/HighFidelity-Beta-PR15325-68c8d2e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18588/HighFidelity-Beta-Interface-PR15325-68c8d2e.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18588/HighFidelity-Beta-PR15325-68c8d2e.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18588/HighFidelity-Quest-Beta-PR15325-68c8d2e.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18588/HighFidelity-Beta-PR15325-68c8d2e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18588/HighFidelity-Beta-Interface-PR15325-68c8d2e.exe

File: scripts/system/html/js/entityProperties.js [hunk 1]
```diff
@@ -3058,24 +3092,148 @@ function setDropdownValue(event) {
...
+    }
+    
+    updateProperty("parentMaterialName", materialTargetList, false);
+}
+
+function materialTargetPropertyUpdate(propertyValue) {
+    let elDivOptions = getPropertyInputElement("parentMaterialName");
+    let elInputs = elDivOptions.getElementsByClassName("materialTargetInput");
+    
+    if (propertyValue.charAt(0) === '[') {
```
Comment (thoys) [inline scripts/system/html/js/entityProperties.js]: `propertyValue.startsWith('[')`   would work here

File: scripts/system/html/js/entityProperties.js [hunk 2]
```diff
@@ -3058,24 +3092,148 @@ function setDropdownValue(event) {
...
+        propertyValue = propertyValue.substring(1, propertyValue.length);
+    }
+    if (propertyValue.charAt(propertyValue.length - 1) === ']') {
```
Comment (thoys) [inline scripts/system/html/js/entityProperties.js]: `propertyValue.endsWith(']')`   would work here

File: scripts/system/html/js/entityProperties.js [hunk 3]
```diff
@@ -3058,24 +3092,148 @@ function setDropdownValue(event) {
...
+function setMaterialTargetData(materialTargetData) {
+    let elDivOptions = getPropertyInputElement("parentMaterialName");
+    resetDynamicMultiselectProperty(elDivOptions);
+    
+    if (materialTargetData === undefined) {
+        return;
     }
+    
+    elDivOptions.firstChild.style.display = "none"; // hide "No Options" text
+    elDivOptions.parentNode.lastChild.style.display = "block"; // show Select/Clear all buttons
```
Comment (thoys) [inline scripts/system/html/js/entityProperties.js]: could set display to `null` to make it fall back to the default display

File: scripts/system/edit.js [hunk 4]
```diff
@@ -2522,7 +2522,19 @@ var PropertiesTool = function (opts) {
                 type: 'propertyRangeReply',
                 propertyRanges: propertyRanges,
             });
-        }
+        } else if (data.type === "materialTargetRequest") {
```
Comment (dback2) [inline scripts/system/edit.js]: Need to follow up with protection on this, ensuring request received matches last request sent

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18617/HighFidelity-Beta-PR15325-f2aa155.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18617/HighFidelity-Beta-Interface-PR15325-f2aa155.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18617/HighFidelity-Beta-PR15325-f2aa155.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18617/HighFidelity-Beta-Interface-PR15325-f2aa155.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18617/HighFidelity-Beta-PR15325-f2aa155.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18617/HighFidelity-Quest-Beta-PR15325-f2aa155.apk)

**Comment by dback2** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18628/HighFidelity-Beta-PR15325-f2aa155.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18628/HighFidelity-Quest-Beta-PR15325-f2aa155.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18628/HighFidelity-Beta-PR15325-f2aa155.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18628/HighFidelity-Beta-Interface-PR15325-f2aa155.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18628/HighFidelity-Beta-PR15325-f2aa155.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18628/HighFidelity-Beta-Interface-PR15325-f2aa155.exe

**Comment by dback2** [comment]
Comments addressed in https://github.com/highfidelity/hifi/pull/15352

**Pull request merged**